### PR TITLE
[NP-6741] Wizard views update

### PR DESCRIPTION
**Pull request by KernelDeimos**
This PR adds the following
- new controllers to (eventually) separate "stepping" logic from general wizard functionality
- early work on "focus wizard" (a particular wizard view)

This PR should not affect any production wizards (i.e. if it does, that's a bug)

### CODE REVIEW

File: src/foam/u2/wizard/views/ProgressBarWizardView.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+foam.CLASS({
```
Comment (jlhughes) [inline src/foam/u2/wizard/views/ProgressBarWizardView.js:7]: header please.

File: src/foam/u2/detail/FlexSectionedDetailView.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.detail',
+  name: 'FlexSectionedDetailView',
+  extends: 'foam.u2.detail.AbstractSectionedDetailView',
+
+  requires: [
+    'foam.u2.detail.SectionView',
```
Comment (kgrgreer) [inline src/foam/u2/detail/FlexSectionedDetailView.js:14]: sort

File: src/foam/u2/wizard/controllers/Form.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      name: 'view',
+      documentation: `
+        Represents the desired view to render the current wizard contents.
+        Form subclasses can choose how they want to use this.
+      `,
+      value: {
+        class: 'foam.u2.borders.NullBorder'
+      }
+    }
+  ],
```
Comment (kgrgreer) [inline src/foam/u2/wizard/controllers/Form.js]: No trailing ,

File: src/foam/u2/wizard/controllers/Form.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.wizard.controllers',
+  name: 'Form',
+  extends: 'foam.u2.Controller',
+
+  issues: [
```
Comment (kgrgreer) [inline src/foam/u2/wizard/controllers/Form.js:12]: !

File: src/foam/u2/wizard/views/FlexibleWizardFormView.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  methods: [
+    function render() {
+      const current$ = this.slot(function (data, data$currentWizardlet, data$currentSection) {
+        return data$currentSection.createView();
+      })
+      this.addClass()
+        .add(current$)
+        .start()
+          .addClass(this.myClass('flexButtons'))
+          .startContext({ data: this.data })
```
Comment (kgrgreer) [inline src/foam/u2/wizard/views/FlexibleWizardFormView.js]: Is this necessary? Isn't data already exported?

File: src/foam/u2/wizard/controllers/Form.js (hunk 4 above)
Comment (KernelDeimos) [reply src/foam/u2/wizard/controllers/Form.js:12]: I'm not sure how to address this comment

File: src/foam/u2/wizard/views/FlexibleWizardFormView.js (hunk 5 above)
Comment (KernelDeimos) [reply src/foam/u2/wizard/views/FlexibleWizardFormView.js]: thanks, forgot that after moving it from a controller

**Pull request merged**